shardy/dialect/sdy/ir/attrs.td

```diff
@@ -176,8 +177,8 @@ def Sdy_AxisRef : AttrDef<Sdy_Dialect, "AxisRef"> {
   let mnemonic = "axis_ref";
   let summary = "Reference to either a full axis or a split sub-axis";
   let parameters = (ins
-      StringRefParameter<"name">:$name,
-      OptionalParameter<"SubAxisInfoAttr">:$sub_axis_info
+      StringRefParameter<"name", "the name of this axis">:$name,
```

Comment:
Turns out this isn't actually a description but a default value, you need to change "name" itself.